src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *                         type: string
+ *                         format: date-time
+ *                         description: The creation timestamp of the post.
+ *                       creator_id:
+ *                         type: object
+ *                         properties:
+ *                           _id:
+ *                             type: string
+ *                             description: The unique identifier of the creator.
+ *                           username:
+ *                             type: string
+ *                             description: The username of the post creator.
+ *                       likes:
+ *                         type: array
+ *                         items:
+ *                           type: object
+ *                           properties:
+ *                             _id:
+ *                               type: string
+ *                               description: The unique identifier of the like.
+ *                             user_id:
+ *                               type: string
+ *                               description: The ID of the user who liked the post.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message
+ */
 router.get('/search', searchPosts);
+
+/**
+ * @swagger
+ * /posts/checkLikes/{postID}:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Count the number of likes for a specific post
+ *     description: Retrieve the total number of likes for a post identified by its ID.
+ *     parameters:
+ *       - in: path
+ *         name: postID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post.
+ *     responses:
+ *       200:
+ *         description: The number of likes for the post.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 count:
+ *                   type: integer
+ *                   description: The total number of likes.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message.
+ *                 err:
+ *                   type: string
+ *                   description: Details of the error.
+ */
 router.get('/checkLikes/:postID', countPostLikes);
+
+/**
+ * @swagger
+ * /posts/topThreeLikes/{postID}:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Retrieve the top three likes for a specific post
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"